megaparsec-tests/tests/Text/Megaparsec/Byte/MachineIntsSpec.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Text.Megaparsec.Byte.MachineIntsSpec (spec) where
+
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as B
+import Data.Void
+import Data.Word
+import Test.Hspec
+import Test.Hspec.Megaparsec
+--import Test.QuickCheck
+import Text.Megaparsec
+import Text.Megaparsec.Byte.MachineInts
+
+-- TODO: use QuickCheck
+
+type Parser = Parsec Void ByteString
+
+spec :: Spec
+spec = do
+  describe "anyLE" $ do
+    it "parses a little-endian Word32" $ do
+      let p = anyLE @Word32 <* eof
+          (s, _, valueLE) = exW32
+      prs p s `shouldParse` valueLE
+      prs' p s `succeedsLeaving` ""
```

Comment:
In this case `succeedsLeaving` is redundant because you have `eof` in your parser. `shouldParser` will never succeed unless it consumes all input. Indeed, QuickCheck could be useful here. A typical test could be:

* Generate a number.
* Serialize it.
* Parse it and compare with the originally generated number.